clients/src/test/java/org/apache/kafka/clients/consumer/internals/FetcherTest.java

```diff
@@ -148,6 +150,33 @@ public boolean matches(ClientRequest request) {
     }
 
     @Test
+    public void testFetchedRecordsRaisesOnSerializationErrors() {
+        ByteArrayDeserializer deserializer = new ByteArrayDeserializer() {
+            @Override
+            public byte[] deserialize(String topic, byte[] data) {
+                throw new SerializationException();
```

Comment:
This always fails on the first message in a fetch response. What is the expected behavior if we fail half way through a batch? I'm curious because you actually received some data which could potentially be returned to the user. It's not great to just throw the exception anyway because then the user doesn't actually know which message caused the problem -- their current position may not be the position of the message causing the exception.

So a few things. First, you might want a test case that validates the behavior when a serialization exception happens on message > 1 of the batch, in particular that the position is what you expect it to be in that case. Second, can we provide the user more information? Either a) by returning partial fetch responses and moving the position up to the offset of the bad message or b) maybe by catching the `SerializationException` and wrapping it or something so you can pass along information about the (topic, partition, offset) that caused the problem?